### PR TITLE
UI support for pipeline notification config

### PR DESCRIPTION
**Pull request by n313893254**
Proposed changes
======
UI support for pipeline notification config
- After:
image

Types of changes
======
- New feature (non-breaking change which adds functionality)

Linked Issues
======
 
https://github.com/rancher/rancher/issues/15666

Further comments
======

### CODE REVIEW

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import Component from '@ember/component';
+import layout from './template'
+import { get, set, observer, computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { reads } from '@ember/object/computed';
+import C from 'ui/utils/constants';
+
+const CONDICTION_CHOICES = [C.CONDICTION.SUCCESS, C.CONDICTION.CHANGED, C.CONDICTION.FAILED]
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
const CONDITION_CHOICES = [C.CONDITION.SUCCESS, C.CONDITION.CHANGED, C.CONDITION.FAILED]
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+export default Component.extend({
+  globalStore: service(),
+  scope:            service(),
+  layout,
+
+  config:     null,
+
+  conditionContent: CONDICTION_CHOICES,
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
  conditionContent: CONDITION_CHOICES,
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    get(this, 'globalStore').findAll('notifier').then((res) => {
+      set(this, 'notifiers', res.filterBy('clusterId', clusterId))
+    })
+    if ( config ) {
+      const { condition = [] } = config
+
+      condition.map((c) => {
+        switch (c) {
+        case C.CONDICTION.SUCCESS:
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
        case C.CONDITION.SUCCESS:
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          set(this, 'success', true)
+          break;
+        case C.CONDICTION.CHANGED:
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
        case C.CONDITION.CHANGED:
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+          set(this, 'changed', true)
+          break;
+        case C.CONDICTION.FAILED:
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
        case C.CONDITION.FAILED:
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+          set(this, 'failed', true)
+          break;
+        default: break;
+        }
+      })
+    }
+    this.condictionChange()
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
    this.conditionChange()
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      set(this, 'failed', false)
+      set(this, 'config.message', null)
+      set(this, 'config.recipients', [{
+        notifierId: null,
+        recipient:  null,
+      }])
+    },
+  },
+
+  condictionChange: observer('success', 'changed', 'failed', function() {
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
  conditionChange: observer('success', 'changed', 'failed', function() {
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    const arr = [
+      get(this, 'success') ? C.CONDICTION.SUCCESS : null,
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
      get(this, 'success') ? C.CONDITION.SUCCESS : null,
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+      get(this, 'changed') ? C.CONDICTION.CHANGED : null,
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
      get(this, 'changed') ? C.CONDITION.CHANGED : null,
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+      get(this, 'failed') ? C.CONDICTION.FAILED : null,
```
Comment (westlywright) [inline lib/pipeline/addon/components/pipeline-notifier/component.js]:
```suggestion
      get(this, 'failed') ? C.CONDITION.FAILED : null,
```

File: lib/shared/addon/utils/constants.js [hunk 11]
```diff
@@ -698,4 +698,10 @@ C.NOTIFIER_TABLE_LABEL = {
 
 C.EXPERIMENTAL_VERSIONS = { RKE_K8S: 'v1.12.0-rancher1-1' }
 
+C.CONDICTION = {
```
Comment (westlywright) [inline lib/shared/addon/utils/constants.js]:
```suggestion
C.CONDITION = {
```

File: lib/pipeline/addon/components/pipeline-notifier/component.js [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+  success:          false,
+  changed:          false,
+  failed:           false,
+  customizeMessage: false,
+
+  clusterId:   reads('scope.currentCluster.id'),
+  init() {
+    this._super(...arguments);
+    const clusterId = get(this, 'clusterId');
+    const config = get(this, 'config');
```
[the remark this answers is not in the document]
Comment (n313893254) [reply lib/pipeline/addon/components/pipeline-notifier/component.js]: cool

File: lib/pipeline/app/components/form-recipient-item/component.js [hunk 13]
```diff
@@ -0,0 +1 @@
+export { default } from 'pipeline/components/form-recipient-item/component';
```
[the remark this answers is not in the document]
Comment (n313893254) [reply lib/pipeline/app/components/form-recipient-item/component.js]: Got it. Thank you for explaining.

File: lib/pipeline/addon/components/pipeline-notifier/template.hbs [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+      {{#if customizeMessage}}
+        {{input value=config.message
+                placeholder=(t 'pipelineNotification.message.placeholder')
+        }}
+      {{/if}}
+    </div>
+  </div>
+
+  {{#unless haveNotifiers}}
+    <div class="banner bg-warning">
```
[the remark this answers is not in the document]
Comment (n313893254) [reply lib/pipeline/addon/components/pipeline-notifier/template.hbs]: @westlywright The message includes `{{#link-to-external}}` to link to create notifiers.  Can `{{banner-message}}` includes `{{#link-to-external}}` ?

[the remark this answers is not in the document]
Comment (westlywright) [reply lib/pipeline/addon/components/pipeline-notifier/template.hbs]: @n313893254 Shouldn't be a problem, its just a component in a component

**Comment by n313893254** [comment]
@westlywright Done, Thanks for your review.

**Pull request merged**